ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/log/IncrementalLoadLogger.java

```diff
@@ -33,6 +33,7 @@
   private String dumpDir;
   private long numEvents;
   private long eventSeqNo;
+  private long currentEventTimestamp;
```

Comment:
The intention was to maintain the timestamp in barrier tasks. The barrier task gets executed once all the tasks for one event finish. So the difference between consecutive barrier-timestamps gives duration for one event plus additional time of processing the previous barrier. In this case, the logger (ReplStateLogTask) happens to be the barrier task so it is maintained here.

Reply:
Is there some other recommendation ?

Reply:
The concern here is that the timing for first event is not accurate. It may be quite a while from the point where ReplLoadWork (which is the start time for first event in your patch) till it actually get started. And additionally this will also impact the other metrics which may add on per type basis in future.